entity_management/core.py

```diff
@@ -433,6 +437,19 @@ class WorkflowExecution(Activity):
     """
 
 
+class GenerationActivity(Activity):
```

Comment:
When I said something more specific, I had in mind a class that derives from Activity and is specific to what generates that AnalysisReport.

Maybe AnalysisReportGeneration? In addition it should not be in core.py but in a module where the rest of the analysis entities are defined.

This is also a good opportunity to move all the entities from bbp-workflow here so that they are all together.